app/assets/javascripts/discourse/app/templates/activate-account.gjs

```diff
@@ -0,0 +1,141 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import RouteTemplate from "ember-route-template";
+import DButton from "discourse/components/d-button";
+import hideApplicationHeaderButtons from "discourse/helpers/hide-application-header-buttons";
+import hideApplicationSidebar from "discourse/helpers/hide-application-sidebar";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { wavingHandURL } from "discourse/lib/waving-hand-url";
+import i18n from "discourse-common/helpers/i18n";
+import I18n from "discourse-i18n";
+
+export default RouteTemplate(
+  class extends Component {
+    @service siteSettings;
+    @service router;
+
+    @tracked accountActivated = false;
+    @tracked buttonDisabled = false;
+    @tracked needsApproval = false;
+    @tracked errorMessage = null;
+
+    get siteName() {
+      return this.siteSettings.title;
+    }
+
+    get wavingHandURL() {
+      return wavingHandURL();
+    }
+
+    get translatedButtonLabel() {
+      return I18n.t("user.activate_account.continue_button", {
+        site_name: this.siteName,
+      });
+    }
+
+    @action
+    async activate() {
+      this.buttonDisabled = true;
```

Comment:
generally speaking, it's better to use a state here, eg: loading. Here you are dictating the implementation, what if actually we decide to use a link later? or if we want to use this at two different places? Also "buttonDisabled" doesn't tell us much on the reason, whereas "isLoading" is more clear

Reply:
Ah sorry, I just remembered it's a port to gjs, not really your own code

Reply:
nope, that part it is, it's a port from a view/script -> template/controller -> gjs